### DISCUSSION TITLE
Example for Multi-Task Learning

### DISCUSSION THREAD

**Discussion by hatala91** [pyg-team/pytorch_geometric#3966] [Ideas]
Hello 👋

I am wondering if y'all think an example for Multi-Task Learning using PyG would be a good addition? I found only a few discussions and issues regarding MTL so I thought I'd ask before drafting up a PR.

The background to this is that I had a problem I tried solving first with a semi-supervised learning and then also tried with an Autoencoder. After trying different convolutional layers, adding dropout, regularization, BN, etc. I next build "one model to rule them all" and was pleasantly surprised by the performance gain.
In a nutshell it's the input layer, than a set of (can also be 0) hidden GCNConvs and finally one head per task (I had two for classification and one autoencoder).

Happy to hear you opinions and what Dataset from the examples this could apply to best 🚀 

P. S. I am a huge fan of PyTorch Geometric 🙏

**Comment by rusty1s** [comment]
Definitely would love to integrate such an example. Please feel free to go ahead :)

**Comment by wsad1** [comment]
There are a few multitask datasets you could use in pyg, in [torch_geometric.datasets](https://pytorch-geometric.readthedocs.io/en/latest/modules/datasets.html#torch-geometric-datasets) like `PPI`, `Yelp`, `QM7b` or `QM9`.

**Comment by ghost** [comment]
Thanks for the feedback 🙏 I added this PR here: https://github.com/pyg-team/pytorch_geometric/pull/4068
